xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/MacroBlock.java

```diff
@@ -128,29 +83,16 @@ public boolean equals(Object obj)
             return true;
         }
 
-        if (obj instanceof MacroBlock && super.equals(obj)) {
-            EqualsBuilder builder = new EqualsBuilder();
-
-            builder.append(getContent(), ((MacroBlock) obj).getContent());
-            builder.append(getId(), ((MacroBlock) obj).getId());
-            builder.append(isInline(), ((MacroBlock) obj).isInline());
-
-            return builder.isEquals();
+        if (!(obj instanceof MacroBlock)) {
+            return false;
         }
 
-        return false;
+        return abstractBlockEquals((MacroBlock) obj);
     }
 
     @Override
     public int hashCode()
```

Comment:
Why not implementing hashCode at abstract level since there is nothing specific here.